src/web/services/auroraApiClients/applicationDeploymentClient/utils.ts

```diff
@@ -12,12 +12,30 @@ export function changeVersionInFile(
       null,
       2
     );
+  } else if (fileName.endsWith('yaml')) {
+    try {
+      return parseYaml(fileContent, version);
+    } catch {
+      return attemptFallbackMethod(fileContent, version);
+    }
   }
+  return undefined;
+}
+
+function attemptFallbackMethod(fileContent: string, version: string) {
   try {
-    const parsedApplicationFile = YAML.parseDocument(fileContent);
-    parsedApplicationFile.set('version', version);
-    return parsedApplicationFile.toString();
+    let emptyYamlWithVersion = fileContent + `\nversion: ${version}`;
+    if (fileContent.endsWith('\n')) {
+      emptyYamlWithVersion = fileContent + `version: ${version}`;
+    }
+    return parseYaml(emptyYamlWithVersion, version);
   } catch {
     return undefined;
   }
 }
+
+function parseYaml(fileContent: string, version: string) {
```

Comment:
Bare for å ta en litt annen vri, kan logikken bli inkludert i parseYaml? Det ville komprimert det en del:

```
  } else if (fileName.endsWith('yaml')) {
    try {
      return parseYaml(fileContent, version);
    } catch {
      return undefined
    }
  }
  return undefined;
}

function ensureEndsWithNewline(value: string) {
  if (value.endsWith('\n')) {
    return value;
  }

  return `${value}\n`;
}

function parseYaml(fileContent: string, version: string) {
  const fileContentWithVersion =
    ensureEndsWithNewline(fileContent) + `version: ${version}`;
  const parsedApplicationFile = YAML.parseDocument(fileContentWithVersion);
  return parsedApplicationFile.toString();
}
```

Reply:
Ja mulig man kan få løst det på en slik måte.
Men det er to problemer med akkurat den løsningen:

- Dersom applikasjonen allerede har version satt i applikasjons fila, så vil det bli dobbelt opp.
- Den `set` metoden som brukes nå for å sette versjon passer på at hvis det legges inn f.eks. major versjon 1 så blir den lagt til med fnutter. Hvis det er en bugfix versjon så blir den lagt til uten fnutter. Akkuratt slik det fungerer i AO. Så hvis man ikke bruker denne metoden, så vil versjon alltid legges på uten fnutter og blir et problemer med major versjon som må ha det.

Reply:
En ting jeg har tenkt på litt for å løse det på denne måten, altså slippe å ha 2 try catch.
Er å gjøre en sjekk tidlig på om `fileContent.trim() === "---"` da vil man vel få håndtert akkurat det caset med tom yaml fil.

Reply:
Endret litt på forslaget ditt, og da kan dette muligens være en grei løsning:
```
  } else if (fileName.endsWith('yaml')) {
    try {
      if (fileContent.trim() === '---') {
        const emptyYamlWithVersion =
          ensureEndsWithNewline(fileContent) + `version: ${version}`;
        return parseYaml(emptyYamlWithVersion, version);
      }
      return parseYaml(fileContent, version);
    } catch {
      return undefined;
    }
  }
  return undefined;
}

function ensureEndsWithNewline(value: string) {
  if (value.endsWith('\n')) {
    return value;
  }

  return `${value}\n`;
}
```

Reply:
Er sikkert litt pirkete nå, men men.
Jeg ser at mitt opprinnelige forslag ikke tok høyde for de tingene du påpekte. 

Hva med dette?

```
const isEmptyYaml = (content: string) => content.trim() === '---';

const addVersionToEmptyYaml = (content: string, version: string) =>
  ensureEndsWithNewline(content) + `version: ${version}`;

function parseYaml(fileContent: string, version: string) {
  const yamlContent = isEmptyYaml(fileContent)
    ? addVersionToEmptyYaml(fileContent, version)
    : fileContent;

  const parsedApplicationFile = YAML.parseDocument(yamlContent);
  return parsedApplicationFile.toString();
}
```

Reply:
Ser bra ut!